modules/xnetevents.php

```diff
@@ -244,10 +244,33 @@ function handler_sub($page, $eid = null)
 
         //if we arrived here via a confirmation email, process it
         if(Get::has('nom') && Get::has('prenom') && Get::has('email') && Get::has('token')){
-            if(Get::v('token')!=make_confirmation_token($nom,$prenom,$email)){
+            if(Get::v('token')!=$this->make_confirmation_token(Get::v('nom'),Get::v('prenom'),Get::v('email'))){
                 $page->kill('Ce lien n\'est pas valide.');
             }
-            XDB::execute('INSERT INTO accounts SET firstname={?}, lastname={?}, hruid={?}, email={?}', Get::v('prenom'), Get::v('nom'), Get::v('prenom').".".Get::v('nom'), Get::v('email'));
+
+            //check if the email address is already registered
+            $res=XDB::query("SELECT hruid FROM accounts WHERE email={?}", Get::v('email'));            
+
+            if($res->numRows()>0){
+                //the email address is already registered, use the registered account
+                $hruid=(string)$res->fetchOneCell();
+            }else{
+                //the email address is not registered, create a new user
+                require_once 'name.func.inc.php';
+                list($local_part, $domain) = explode('@', strtolower(Get::v('email')));
+                $firstname=Get::v('prenom');
+                $lastname=Get::v('nom');
+                $hruid = User::makeHrid($local_part, $domain, 'ext');
+                $full_name = build_full_name($firstname, $lastname);
+                $directory_name = build_directory_name($firstname, $lastname);
+                $sort_name = build_sort_name($firstname, $lastname);
+                XDB::execute('INSERT INTO accounts SET firstname={?}, lastname={?}, full_name={?}, directory_name={?}, sort_name={?}, hruid={?}, email={?}, type={?}, state={?}', $firstname, $lastname, $full_name, $directory_name, $sort_name, $hruid, Get::v('email'), "xnet", "active");
```

Comment:
This may break if an hruid already exists for the given email address? ``User::makeHrid`` (https://github.com/Polytechnique-org/platal/blob/core/1.1.16/maint/classes/pluser.php#L431) is not injective. For example:

    makeHrid("example++test@example.org") == makeHrid("example+test@example.org")

A possible solution would be to add a counter when a collision happens